persistence/src/main/scala/akka/persistence/EventStoreinterop.scala

```diff
@@ -0,0 +1,157 @@
+package akka.persistence
+
+import akka.actor.ActorSystem
+
+import cats.syntax.all._
+import cats.effect.{Async, Sync, Ref}
+
+import com.evolutiongaming.akkaeffect.ActorEffect
+import com.evolutiongaming.akkaeffect.persistence.{EventStore, EventSourcedId, SeqNr, Events}
+import com.evolutiongaming.catshelper.{FromFuture, ToTry}
+import com.evolutiongaming.sstream
+
+import scala.concurrent.duration._
+
+object EventStoreInterop {
+
+  def apply[F[_]: Async: FromFuture: ToTry, A](
+    system: ActorSystem,
+    timeout: FiniteDuration,
+    journalPluginId: String,
+    eventSourcedId: EventSourcedId
+  ): F[EventStore[F, A]] =
+    Sync[F]
+      .delay {
+        val actorRef = Persistence(system).journalFor(journalPluginId)
+        ActorEffect.fromActor(actorRef)
+      }
+      .map { journaller =>
+        new EventStore[F, A] {
+
+          import sstream.FoldWhile._
+
+          val persistenceId = eventSourcedId.value
+
+          override def events(fromSeqNr: SeqNr): F[sstream.Stream[F, EventStore.Persisted[A]]] = {
+
+            type Buffer = Vector[EventStore.Event[A]]
+
+            def actor(buffer: Ref[F, Buffer]) =
+              LocalActorRef[F, Unit, SeqNr]({}, timeout) {
+
+                case (_, JournalProtocol.ReplayedMessage(persisted)) =>
+                  if (persisted.deleted) ().asLeft[SeqNr].pure[F]
+                  else {
+                    val payload = persisted.payload.asInstanceOf[A]
+                    val event   = EventStore.Event(payload, persisted.sequenceNr)
+                    buffer.update(_ :+ event).as(().asLeft[SeqNr])
```

Comment:
out of memory? :)

Reply:
https://github.com/evolution-gaming/akka-effect/pull/276#discussion_r1446394666